source/common/buffer/buffer_impl.h

```diff
@@ -764,24 +756,20 @@ class OwnedImpl : public LibEventInstance {
   uint8_t* inlineReserve(uint64_t size) override {
     if (slices_.empty()) {
       slices_.emplace_back(Slice(size, account_));
-      return slices_.back().inlineReserve(size);
+      length_ += size;
+      slices_.back().reservable_ += size;
+      return slices_.back().base_;
     }
 
-    auto* mem = slices_.back().inlineReserve(size);
-    if (mem != nullptr) {
+    auto& back = slices_.back();
+    if (slices_.back().reservableSize() >= size) {
+      uint8_t* mem = back.base_ + back.reservable_;
+      length_ += size;
+      back.reservable_ += size;
       return mem;
     }
 
-    slices_.emplace_back(Slice(size, account_));
-    return slices_.back().inlineReserve(size);
-  }
-
-  void inlineCommit(uint64_t size) override {
-    ASSERT(!slices_.empty());
-    slices_.back().reservable_ += size;
-    ASSERT(slices_.back().reservable_ <= slices_.back().capacity_);
-
-    length_ += size;
+    return nullptr;
```

Comment:
can you comment on the decision process to return null here, which will need to be checked elsewhere? Will the caller have to do something with this null-return that we couldn't figure out and just do it here?

Reply:
`inlineReserve` is a protected method and only be used by the `addFragments`. null-return means that there is not enough continuous memory in the back slice, `addFragments` will downgrade to use classical `add`. The related descriptions are added in the return description and method description.

Reply:
😄 I also added some comments in the code.

Reply:
thanks!